pkg/providers/gcp/provider_postgres.go

```diff
@@ -265,7 +293,7 @@ func (p *PostgresProvider) deleteCloudSQLInstance(ctx context.Context, networkMa
 		}
 		if !foundInstance.Settings.DeletionProtectionEnabled {
 			_, err = sqladminService.DeleteInstance(ctx, strategyConfig.ProjectID, foundInstance.Name)
-			if err != nil && !resources.IsNotFoundError(err) {
+			if err != nil && !resources.IsNotStatusConflictError(err) {
 				msg := fmt.Sprintf("failed to delete postgres instance: %s", cloudSQLDeleteConfig.Name)
 				return croType.StatusMessage(msg), errorUtil.Wrap(err, msg)
 			}
```

Comment:
You could do something like that here. It reads easier and doesn't pollute the logs in case we attempt to delete while the instance is being upgraded (conflict error):
```
if foundInstance.Settings.DeletionProtectionEnabled {
	update := &sqladmin.DatabaseInstance{
		Settings: &sqladmin.Settings{
			ForceSendFields: []string{"DeletionProtectionEnabled"}, DeletionProtectionEnabled: false},
	}
	_, err := sqladminService.ModifyInstance(ctx, strategyConfig.ProjectID, foundInstance.Name, update)
	if err != nil && !resources.IsConflictError(err) {
		msg := fmt.Sprintf("failed to disable deletion protection for cloudsql instance: %s", foundInstance.Name)
		return croType.StatusMessage(msg), errorUtil.Wrap(err, msg)
	}
	msg := fmt.Sprintf("disabling deletion protection for cloudsql instance %s", foundInstance.Name)
	return croType.StatusMessage(msg), nil
}
_, err = sqladminService.DeleteInstance(ctx, strategyConfig.ProjectID, foundInstance.Name)
if err != nil && !resources.IsNotFoundError(err) {
	msg := fmt.Sprintf("failed to delete postgres instance: %s", foundInstance.Name)
	return croType.StatusMessage(msg), errorUtil.Wrap(err, msg)
}
```